### PR TITLE
VEN-109 Add Names to Users and in APIs

### PR DESCRIPTION
**Pull request by aleksandarpetrushev**
https://getvendo.atlassian.net/browse/VEN-109

### CODE REVIEW

File: api/spec/requests/spree/api/v2/storefront/account_spec.rb [hunk 1]
```diff
@@ -91,6 +93,8 @@
         password_confirmation: 'newpassword123',
         bill_address_id: default_bill_address.id.to_s,
         ship_address_id: default_ship_address.id.to_s,
+        first_name: 'Joe',
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/account_spec.rb:96]: @aleksandarpetrushev let's use only fictional people names :)

Comment (aleksandarpetrushev) [reply api/spec/requests/spree/api/v2/storefront/account_spec.rb:96]: Sure, also need to add docs

**Review by damianlegawiec — APPROVED** [review]

**Review by damianlegawiec — APPROVED** [review]

File: core/db/migrate/20220201103922_add_first_name_and_last_name_to_spree_users.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddFirstNameAndLastNameToSpreeUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_column :spree_users, :first_name, :string unless column_exists?(:spree_users, :first_name)
+    add_column :spree_users, :last_name, :string unless column_exists?(:spree_users, :last_name)
```
Comment (damianlegawiec) [inline core/db/migrate/20220201103922_add_first_name_and_last_name_to_spree_users.rb:4]:
```suggestion
    if Spree.user_class.present?
      add_column Spree.user_class.table_name, :first_name, :string unless column_exists?(:spree_users, :first_name)
      add_column Spree.user_class.table_name, :last_name, :string unless column_exists?(:spree_users, :last_name)
    end
```

File: core/db/migrate/20220201103922_add_first_name_and_last_name_to_spree_users.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class AddFirstNameAndLastNameToSpreeUsers < ActiveRecord::Migration[5.2]
+  def change
+    if Spree.user_class.present?
+      add_column :spree_users, :first_name, :string unless column_exists?(:spree_users, :first_name)
```
Comment (damianlegawiec) [inline core/db/migrate/20220201103922_add_first_name_and_last_name_to_spree_users.rb:4]:
```suggestion
      add_column Spree.user_class.table_name, :first_name, :string unless column_exists?(:spree_users, :first_name)
```

the table name can be different than spree_users

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: core/db/migrate/20220201103922_add_first_name_and_last_name_to_spree_users.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class AddFirstNameAndLastNameToSpreeUsers < ActiveRecord::Migration[5.2]
+  def change
+    if Spree.user_class.present?
+      add_column Spree.user_class.table_name, :first_name, :string unless column_exists?(:spree_users, :first_name)
```
Comment (damianlegawiec) [inline core/db/migrate/20220201103922_add_first_name_and_last_name_to_spree_users.rb:4]:
```suggestion
      user_table_name = Spree.user_class.table_name
      add_column user_table_name, :first_name, :string unless column_exists?(user_table_name, :first_name)
```

Comment (damianlegawiec) [reply core/db/migrate/20220201103922_add_first_name_and_last_name_to_spree_users.rb:4]: @aleksandarpetrushev

**Pull request merged**